src/main/java/org/terasology/gooeyDefence/health/HealthComponent.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.gooeyDefence.health;
+
+import org.terasology.entitySystem.Component;
+
+/**
+ * Adds health functionality to an entity.
+ */
+public class HealthComponent implements Component {
+    private int health;
+
+    /**
+     * Reduce the health by a given amount.
+     *
+     * @param damage The amount to reduce the health by.
+     */
+    public void dealDamage(int damage) {
```

Comment:
No logic in components please :-) Violates one of the core tenets of our ES architecture (and will be prevented entirely with `gestalt-entity-system` IIRC)

Just public fields, nothing else. Logic goes entirely in the system classes.

Reply:
Whoops, will fix.
I thought logic on the level of this was okay as it's directly operating on the data alone.